switchbot/devices/curtain.py

```diff
@@ -63,27 +68,31 @@ async def _send_multiple_commands(self, keys: list[str]) -> bool:
         return final_result
 
     @update_after_operation
-    async def open(self) -> bool:
-        """Send open command."""
-        return await self._send_multiple_commands(OPEN_KEYS)
+    async def open(self, speed="ff") -> bool:
+        """Send open command. Speed 00 or ff - normal, 01 - slow"""
+        return await self._send_multiple_commands(
+            [key + (speed + "00") * (i == 1) for i, key in enumerate(OPEN_KEYS)]
+        )
 
     @update_after_operation
-    async def close(self) -> bool:
-        """Send close command."""
-        return await self._send_multiple_commands(CLOSE_KEYS)
+    async def close(self, speed="ff") -> bool:
+        """Send close command. Speed 00 or ff - normal, 01 - slow"""
+        return await self._send_multiple_commands(
+            [key + (speed + "64") * (i == 1) for i, key in enumerate(CLOSE_KEYS)]
+        )
 
     @update_after_operation
     async def stop(self) -> bool:
         """Send stop command to device."""
         return await self._send_multiple_commands(STOP_KEYS)
 
     @update_after_operation
-    async def set_position(self, position: int) -> bool:
-        """Send position command (0-100) to device."""
+    async def set_position(self, position: int, speed="ff") -> bool:
```

Comment:
Please add a test for these changes

Reply:
There are tests for parsers but not for functions like open/close. Which tests do you mean? The only thing i made was replacing ff flag with speed flag.

*Sorry it's a bit tough for me - i am not a python dev so i struggle a bit